### ISSUE TITLE
Include other options in string?

### ISSUE THREAD

**Issue by ziggie216** [manbearwiz/youtube-dl-server#29]
I noticed that YouTube-dl-server will stop if there is a private video listed YT playlist. Is there a way I can include —ignore-errors

**Comment by ohnggni** [comment]
In order to that options,
you have to modify 'youtube-dl-server.py' as in the following.

1) add the statement, _'YDL_IGNORE_ERR': None_, in line 16 app_default = { below
2) add the statement, _'ignoreerrors': ydl_vars['YDL_IGNORE_ERR']_, in line 101(?) return { below

after that, re-start the docker-container.

<img width="584" alt="스크린샷 2019-03-19 오전 8 05 54" src="https://user-images.githubusercontent.com/33609687/54569368-ef408200-4a1d-11e9-814b-6529dcd94633.png">
<img width="655" alt="스크린샷 2019-03-19 오전 8 06 16" src="https://user-images.githubusercontent.com/33609687/54569374-f5cef980-4a1d-11e9-8f1b-452e1573770c.png">

**Comment by manbearwiz** [comment]
Has anyone tried running with the ignore error flag enabled? Wondering if there is any harm in just enabling it by default.

**Comment by ohnggni** [comment]
> Has anyone tried running with the ignore error flag enabled? Wondering if there is any harm in just enabling it by default. We don't really report any errors back to the user anyway.

I learned this flag does not work in python based on this docker.
If any event (eg. any exception due to request of deleted clips) happen, youtube-dl-server.py is suspended and after that, I can't download any clip. In order to escape this situation, it was the only way to restart the docker container.

I think we need some modification that the python script restarts automatically in case of happening any exception or error.

**Comment by manbearwiz** [comment]
Probably just need to catch and handle errors on that thread. Do you have a playlist you were testing with?

**Comment by ohnggni** [comment]
> Probably just need to catch and handle errors on that thread. Do you have a playlist you were testing with?

https://www.youtube.com/watch?v=wNtbfEaAw18&list=PLofOAS0adMaxoTfZf7HLJ5Vk3As5JuDtE&index=2

The link above is what I was testing. The fourth clip was deleted. So, when I tried to download it, the error happened and the process was suspended at that point. Then I tried downloading another clip(even very normal) but I couldn't do until I restarted the docker container.

**Comment by ohnggni** [comment]
I've just found the solution.
The part of the code above are slightly wrong.
'YDL_IGNORE_ERR': None => True

After that, --ignore-errors works so well and youtube-dl-server.py weren't suspended anymore.

**Comment by manbearwiz** [comment]
Do you think I should make this the default setting?

**Comment by ohnggni** [comment]
Yes, when we try to download playlists, oftentimes we experience that kind of error. Otherwise, we will always have to restart the docker-container whenever we meet any exception or error of python.
